client/my-sites/upgrades/domain-management/dns/txt-record.jsx

```diff
@@ -34,20 +35,26 @@ const TxtRecord = React.createClass( {
 		const classes = classnames( { 'is-hidden': ! this.props.show } ),
 			isValid = this.props.isValid,
 			isNameValid = isValid( 'name' ),
-			isDataValid = isValid( 'data' );
+			isDataValid = isValid( 'data' ),
+			hasNonAsciiData = ! /^[\u0000-\u007f]*$/.test( this.props.fieldValues.data );
```

Comment:
Oh, wow, it took me awhile to understand why this regexp works - I get now that you are trying to verify that all the characters in the string are in the ASCII range. But I think I was influenced by the `hasNonAsciiData` variable name and was fixated on trying to understand it from the "find a non-ASCII data" point of view.
Anyway, how about a simpler regex that does just that:
```
hasNonAsciiData = /[^\u0000-\u007f]/.test( this.props.fieldValues.data );
// vs
hasNonAsciiData = ! /^[\u0000-\u007f]*$/.test( this.props.fieldValues.data );
```
Benefits of this version:
* I think it shows the intent better - literally, test for a non-ASCII character anywhere in the string.
* It should be _marginally_ faster - it doesn't have to verify the whole string (it will finish at the first occurrence of a non-ASCII character).
* It's shorter! And avoids the negation of the `test` result.

Let me know if that makes sense to you too :)